Riot/Modules/MatrixKit/Models/Room/MXKRoomDataSource.m

```diff
@@ -2358,15 +2358,18 @@ - (void)didReceiveReceiptEvent:(MXEvent *)receiptEvent roomState:(MXRoomState *)
 
         // Update cell data we have received a read receipt for
         NSArray *readEventIds = receiptEvent.readReceiptEventIds;
-        for (NSString* eventId in readEventIds)
+        NSArray *readThreadIds = receiptEvent.readReceiptThreadIds;
+        for (int i = 0 ; i < readEventIds.count ; i++)
         {
+            NSString *eventId = readEventIds[i];
             MXKRoomBubbleCellData *cellData = [self cellDataOfEventWithEventId:eventId];
             if (cellData)
             {
+                NSString *threadId = readThreadIds[i] == [NSNull null] ? kMXEventTimelineMain : readThreadIds[i];
```

Comment:
We could simplify this after https://github.com/matrix-org/matrix-ios-sdk/pull/1617/files#r1004308936

Reply:
Unthreaded Res must be propagated to all threads. I've made the needed changes.